### PR TITLE
add the num_freqs arg to .to_source methods

### LINKED ISSUE #1926: `.to_source` methods does not accept the `num_freqs` argument.
A user pointed out that the `.to_source`  method for the ComponentModeler class does not accept the `num_freqs` argument. 
The same goes for the `ModeSolver.to_source`.

It is possible to use the `updated_copy` and to set the `num_freqs`, but maybe it would be good to add this option to the `to_source` methods.

Priority: small

### PR DESCRIPTION
**Pull request by FilipeFcp**
Hi @tylerflex.
I am opening the PR to fix #1926.

I just added the `num_freqs` argument to the `.to_source` method for the `ModeSolve` and `ComponentModeler`.
I added it with the default value of 1.

I added only you as a reviewer only because it seems quite simple.

Thank you

### CODE REVIEW

**Comment by tylerflex** [comment]
Looks great, thanks @FilipeFcp ! 

could you also add a single line under the changelog "fixed" in Unreleased? describing the fix?

Also adding @caseyflex just so he's aware and whether he sees any potential conflicts with the new mode solver

**Comment by tylerflex** [comment]
@FilipeFcp after you add a commit with the changelog edits, you can try using `git rebase` to squash these into a single commit, which helps keep the history more clear.  Will be a good thing to learn as we use it all the time. There's an explanation here

https://www.notion.so/flexcompute/Git-Rebasing-6a7b343ee1cf4ca28fdad0a870354eee#3401e891b41447e3a1defd84c19023cb

**Comment by FilipeFcp** [comment]
Thanks, @tylerflex! I have read that, but I thought that since there is only a single commit, there would be nothing to squash. Is there something I might be missing?

**Comment by tylerflex** [comment]
If you add the change to the changelog , it will add another commit to this PR

**Comment by FilipeFcp** [comment]
@tylerflex Of course, thanks for clarifying! Done.

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Improved passivity enforcement near high-Q poles in `FastDispersionFitter`. Failed passivity enforcement could lead to simulation divergences.
 - More helpful error and suggestion if users try to differentiate w.r.t. unsupported `FluxMonitor` output.
 - Removed positive warnings in Simulation validators for Bloch boundary conditions.
+- Added the option to specify the `num_freqs` argument to the `._to_source` method for both `ModeSolver` and `ComponentModeler`.
```
Comment (tylerflex) [inline CHANGELOG.md]: please remove underscore `_` in `_to_source`

**Review by tylerflex — CHANGES_REQUESTED** [review]
Looks good, thanks @FilipeFcp , just a comment about an extra `_` in the changelog.

**Review by tylerflex — APPROVED** [review]
Thanks @FilipeFcp !

**Comment by FilipeFcp** [comment]
Thank you @tylerflex and @daquinteroflex, for reviewing and taking the time to help me with the workflow.

**Comment by tylerflex** [comment]
quick study :D

**Review by caseyflex — APPROVED** [review]
Thanks for the heads up! Looks good. I guess we could also generally support passing kwargs for this kind of method, but it's good to highlight individual arguments that may be of interest.

I will have to be rather careful about conflicts until my ModeSimulation PR is merged -- it seems like it could be easy to accidentally lose changes made to the ModeSolver plugin, since my PR moves everything to components. I'll have to manually move over every change made to the ModeSolver in the interim.

**Comment by tylerflex** [comment]
Hi @FilipeFcp looks like the changelog got out of date with `develop`, so could you do one last rebase here (against the updated version of develop, eg after performing a git pull origin develop)? And if you're inclined, you could add **kwargs to the method signatures and pass these `**kwargs` in along with the `num_freqs`.

Then this should be good to merge (which I'll take care of)

Thanks

**Comment by FilipeFcp** [comment]
I think is all set @tylerflex. Thanks!

**Pull request merged**